src/libAtomVM/nifs.c

```diff
@@ -5842,6 +5868,118 @@ static term nif_erlang_lists_subtract(Context *ctx, int argc, term argv[])
     return result;
 }
 
+
+static term nif_lists_member(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc)
+    term elem = argv[0];
+    term list = argv[1];
+    VALIDATE_VALUE(list, term_is_list);
+
+    int proper;
+    term_list_length(list, &proper);
+    if (UNLIKELY(!proper)) {
+        RAISE_ERROR(BADARG_ATOM);
+    }
```

Comment:
We can remove this part, so it can be just `O(1)`: instead of using `term_list_length` to check if it is a proper list or not, we can check it at the end.
So we can also be equivalent to the OTP implementation:
```elixir
iex(3)> :lists.member(3, [1, 2, 3 | :foo])
true
iex(4)> :lists.member(4, [1, 2, 3 | :foo])
** (ArgumentError) errors were found at the given arguments:

  * 2nd argument: not a proper list

    (stdlib 6.2) :lists.member(4, [1, 2, 3 | :foo])
    iex:4: (file)
```